internal/options/options.go

```diff
@@ -79,8 +84,9 @@ func (o *Options) AddFlags() {
 	o.flags.IntVar(&o.ExporterPort, "exporter-port", 8081, "Port to expose kube-events-exporter own metrics on.")
 	o.flags.BoolVar(&o.Version, "version", false, "kube-events-exporter version information")
 
-	o.flags.StringArrayVar(&o.EventTypes, "event-types", []string{EventTypesAll}, "List of allowed Event types. Defaults to all types.")
-	o.flags.StringArrayVar(&o.InvolvedObjectAPIGroups, "involved-object-api-groups", []string{APIGroupsAll}, "List of allowed Event involved object API groups. Defaults to all API groups.")
+	o.flags.StringArrayVar(&o.ControllersReportingReasons, "controllers-reporting-reasons", []string{ControllerReportingReasonAll}, "List of controllers allowed to report Event reasons. Filtered-out reasons will be replaced by ReasonMaskedByExporter. Defaults to all controllers.")
```

Comment:
the controllers-reporting-reasons seems a bit awkward flag name. We can do anything here as its not an official API name so lets try to be more descriptive:
--allowed-controllers would be better IMO but up to you.

Reply:
I agree that we definitely need to change this flag name.
My concern with --allowed-controllers is that it might confuse users into thinking that only the event reported by these controllers are allowed while we only wanted to add a mechanism to mask reasons.

However, giving it another thought the current mechanism will be quite hard to grasp for the users and the metrics with masked reason will not be really useful to them. The event reason is the most helpful information when it comes to debugging or even understanding what is really happening so masking it would be as useful as if the metric did not exist in the first place.

Thus, I am thinking of using the "--allowed-controllers" and filter-out the events instead of just masking the reasons. wdyt?

Reply:
> Thus, I am thinking of using the "--allowed-controllers" and filter-out the events instead of just masking the reasons. wdyt?

So this would look like:
````
--allowed-controllers=secrets,configmap
```

so you would not get any other events for other controllers? Would not `involved-object-api-groups ` do the same thing?